### ISSUE TITLE
There are some issues in the Unit Testing part

### ISSUE THREAD

**Issue by hoangnguyen92dn** [hm-tamim/nimble-survey#4]
It’s good to see there are UnitTest for UseCase; besides that, I am also curious about your UnitTest with some points as below:

- [x] As the UseCase test is UnitTest so does it require to execute with RobolectricTestRunner? Can you please clarify a bit more insight about the test here for example?

https://github.com/hm-tamim/nimble-survey/blob/6b084556f855e13bd07da436040cab63eeed6832/app/src/test/java/co/nimblehq/survey/domain/usecase/LoginUseCaseTest.kt#L15-L21

- [x] When I execute the tests case, the "real" API requests are sent to the backend, it seems like “integration” testing instead of “Unit” testing which the execution will be across UseCase > Repository > API Service.

https://github.com/hm-tamim/nimble-survey/blob/6b084556f855e13bd07da436040cab63eeed6832/app/src/test/java/co/nimblehq/survey/domain/usecase/LoginUseCaseTest.kt#L31-L41

- [x] I am wondering why we have to put the `fun beforeClass()` inside `companion object`, would you mind clarifying your idea here?

https://github.com/hm-tamim/nimble-survey/blob/6b084556f855e13bd07da436040cab63eeed6832/app/src/test/java/co/nimblehq/survey/domain/usecase/LoginUseCaseTest.kt#L43-L49

- [x] Furthermore, since the LoginUseCase is located in the `:domain` module, so why the LoginUseCaseTest is located in the `:app` module? Please correct me if I am missing anything?

![image](https://user-images.githubusercontent.com/6950766/162282258-65ffc4b5-f40c-4cd5-b461-7646f3fa8d66.png)

- [x] The test cases in `FragmentTest` are now just validating very simple cases by checking the text. Would it be possible to add more test cases to verify other cases such as binding data on screen or navigation...? It would improve the code coverage much better.

**Comment by hm-tamim** [comment]
For 1st, 2nd, 4th points, it was a mistake in hurry. I've added proper units tests with fake repositories for all the usecases. And moved all the usecase tests to :domain module.

For 3rd, I was having issue running the tests due to EncryptedSharedPreference. Was getting `java.security.KeyStoreException: AndroidKeyStore not found`

A workaround I found to fix this issue was to use a fake AndroidKeyStore. Ref https://encrypted-storage.matthewnelson.io/get_started/
https://github.com/hm-tamim/nimble-survey/blob/e36896e5fb4f00d0df9a2ec45d9b3108e54d6851/app/src/test/java/co/nimblehq/survey/test/FakeAndroidKeyStore.kt#L14-L24
@BeforeClass is static, which executed before you have test object instance. So it's inside companion to setup FakeAndroidKeyStore

Working on 5th task now.

**Comment by hm-tamim** [comment]
Hi @hoangnguyen92dn,

For 1st, 2nd, 4th points, it was a mistake in hurry. I've added proper units tests with fake repositories for all the usecases. And moved all the usecase tests to :domain module.

For 3rd, I was having issue running the tests due to EncryptedSharedPreference. Was getting `java.security.KeyStoreException: AndroidKeyStore not found`

A workaround I found to fix this issue was to use a fake AndroidKeyStore. Ref https://encrypted-storage.matthewnelson.io/get_started/
https://github.com/hm-tamim/nimble-survey/blob/e36896e5fb4f00d0df9a2ec45d9b3108e54d6851/app/src/test/java/co/nimblehq/survey/test/FakeAndroidKeyStore.kt#L14-L24
@BeforeClass is static, which executed before you have test object instance. So it's inside companion to setup FakeAndroidKeyStore

Working on 5th task now.
